### PR TITLE
Describe impact assessment procedure

### PR DESCRIPTION
**Pull request by mixphix**

### CODE REVIEW

**Comment by mixphix** [comment]
Just thought this might be useful to actually write down somewhere.

**Comment by googleson78** [comment]
I believe submitting a [migration guide](https://github.com/haskell/core-libraries-committee/tree/main/guides) is also a recommended contribution when suggesting breaking changes. Or it might even be mandatory?

File: IMPACT-ASSESSMENTS.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Impact assessments
+
+To provide a clearer picture of the downstream effects of a proposal, the CLC requires that proposers draft an impact assessment of their changes before a final decision is reached. An impact assessment MUST be completed if the proposal contains [breaking changes](https://pvp.haskell.org/), and is RECOMMENDED even if the changes are simply additive, especially in cases of anticipated name clashing or when re-exporting from the Prelude.
+
+## The process
+
+- Fork and build [a newer version of GHC](https://gitlab.haskell.org/ghc/ghc#building-installing) (say, the latest official release) that includes the changes made in your proposal. You should also make a merge request with your changes and add it to your proposal's issue.
```
Comment (tomjaguarpaw) [inline IMPACT-ASSESSMENTS.md:7]: I don't understand "a newer version of GHC" and "the latest official release". Taken literally, they seem contradictory to me. Does it mean "take the tagged commit of the latest official GHC release and add to it just the change you are proposing to make"?

Comment (mixphix) [reply IMPACT-ASSESSMENTS.md:7]: Honestly I'm not super clear on the details here. You're correct that that was the meaning I was trying to convey; whether that's "best practice" is another question!

**Comment by mixphix** [comment]
> I believe submitting a [migration guide](https://github.com/haskell/core-libraries-committee/tree/main/guides) is also a recommended contribution when suggesting breaking changes. Or it might even be mandatory?

It doesn't say it's mandatory in `PROPOSALS.md`, but it doesn't say anything at all. It would be a good idea to update that one too. I'll change the bit about compatibility patches to include a blurb about migration guides.

**Comment by Bodigrim** [comment]
@mixphix thanks for writing this! Maybe integrate it with `PROPOSALS.md`?

I've expanded https://github.com/Bodigrim/clc-stackage#how-to to cover necessary technicalities.

**Comment by Bodigrim** [comment]
@mixphix any chance you can integrate this into `PROPOSALS.md` please?

**Comment by Bodigrim** [comment]
Sorry to hear, get well soon!

File: PROPOSALS.md [hunk 2]
```diff
@@ -87,3 +85,14 @@ that the CLC is not doing its due diligence in monitoring active issues,
...
+
+- Fork and build [a newer version of GHC](https://gitlab.haskell.org/ghc/ghc#building-installing) (say, the latest official release) that includes the changes made in your proposal. You should also make a merge request with your changes and add it to your proposal's issue.
+- Try to build [`clc-stackage`](https://github.com/Bodigrim/clc-stackage) using your custom souped-up GHC. If anything breaks, point it out in your update, and notify the packages' maintainers (or raise a PR if you're able!). At least, point the maintainers to your proposal and describe the scope of the breakage.
+- While not strictly the onus of the proposer, we encourage you to submit compatibility patches to broken libraries, to minimize library maintainers' time spent on "`base` churn" ;)
```
Comment (Bodigrim) [inline PROPOSALS.md:95]: There is generally not much point to raise PRs for a yet unapproved proposal, especially if a controversial one. Let's move this step below, like, "If the proposal is approved we encourage you..."

File: PROPOSALS.md [hunk 3]
```diff
@@ -87,3 +85,14 @@ that the CLC is not doing its due diligence in monitoring active issues,
...
 the owners of the `base` library, should have the power to withdraw custodianship
 from the committee.
+
+## Impact assessments
+
+To provide a clearer picture of the downstream effects of a proposal, the CLC requires that proposers draft an impact assessment of their changes before a final decision is reached. An impact assessment MUST be completed if the proposal contains [breaking changes](https://pvp.haskell.org/), and is RECOMMENDED even if the changes are simply additive, especially in cases of anticipated name clashing or when re-exporting from the Prelude.
+
+- Fork and build [a newer version of GHC](https://gitlab.haskell.org/ghc/ghc#building-installing) (say, the latest official release) that includes the changes made in your proposal. You should also make a merge request with your changes and add it to your proposal's issue.
+- Try to build [`clc-stackage`](https://github.com/Bodigrim/clc-stackage) using your custom souped-up GHC. If anything breaks, point it out in your update, and notify the packages' maintainers (or raise a PR if you're able!). At least, point the maintainers to your proposal and describe the scope of the breakage.
```
Comment (Bodigrim) [inline PROPOSALS.md:94]: Notifying maintainers on every unapproved proposal might be too much noise. Historically we have not been demanding this. Let's downgrade it to optional.

See below about raising PRs preemptively.

**Review by Bodigrim — COMMENTED** [review]
Thanks for doing this @mixphix!

**Review by Bodigrim — APPROVED** [review]
Thanks for writing this!

**Comment by Bodigrim** [comment]
Offtopic: @mixphix could you please drop me your up-to-date email address?

**Comment by mixphix** [comment]
I received an email from you today at the correct one.

**Pull request merged**